packages/flutter_tools/lib/src/commands/attach.dart

```diff
@@ -325,19 +294,33 @@ class AttachCommand extends FlutterCommand {
         result = await app.runner.waitForAppToFinish();
         assert(result != null);
```

Comment:
Can you return here and remove the 'else' ?

Reply:
+1